parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -781,6 +910,20 @@ pub mod pallet {
 			});
 
 			Self::deposit_event(Event::BeaconHeaderImported { block_hash: block_root, slot });
+
+			Ok(())
+		}
+
+		fn add_finalized_header_slot(slot: u64) -> DispatchResult {
+			<FinalizedBeaconHeaderSlots<T>>::try_mutate(|b_vec| {
+				if b_vec.len() as u32 == T::MaxFinalizedHeaderSlotArray::get() {
+					b_vec.remove(0);
+				}
+				b_vec.try_push(slot)
+			})
+			.map_err(|_| <Error<T>>::FinalizedBeaconHeaderSlotsExceeded)?;
+
```

Comment:
seems we'are doing some prune work for [FinalizedBeaconHeaderSlots](https://github.com/Snowfork/snowbridge/blob/36c99689aab88a82529931212d72a2157dd8954b/parachain/pallets/ethereum-beacon-client/src/lib.rs#L183), is it also nessessary to prune the related [FinalizedBeaconHeadersBlockRoot](https://github.com/Snowfork/snowbridge/blob/36c99689aab88a82529931212d72a2157dd8954b/parachain/pallets/ethereum-beacon-client/src/lib.rs#L187) storage here?

Reply:
The main reason why I added pruning for `FinalizedBeaconHeaderSlots` is because the beacon relayer queries the whole array, so it needs to be kept in a queryable size. I think we can prune `FinalizedBeaconHeadersBlockRoot` as part of the other pruning ticket: https://linear.app/snowfork/issue/SNO-328